pkg/apis/resource/v1alpha1/pipelineresource_validation.go

```diff
@@ -132,3 +139,15 @@ func validateURL(u, path string) *apis.FieldError {
 	}
 	return nil
 }
+
+func validatePullRequest(s *PipelineResourceSpec) *apis.FieldError {
+	if s.Type != PipelineResourceTypePullRequest {
```

Comment:
We're doing this check twice, once on line 107 and again here. Suggest removing this one.

Reply:
Thank you for the review. Fixed.